newa/__init__.py

```diff
@@ -660,20 +684,32 @@ def id(self) -> str:
 
 
 @define
-class ErratumJob(Job):
+class ArtifactJob(EventJob):
     """ A single *erratum* job """
 
-    erratum: Erratum = field(  # type: ignore[var-annotated]
-        converter=lambda x: x if isinstance(x, Erratum) else Erratum(**x),
+    erratum: Optional[Erratum] = field(  # type: ignore[var-annotated]
+        converter=lambda x: None if x is None else x if isinstance(x, Erratum) else Erratum(**x),
         )
 
+    compose: Optional[Compose] = field(  # type: ignore[var-annotated]
+        converter=lambda x: None if x is None else x if isinstance(x, Compose) else Compose(**x),
+        )
+
+    @property
+    def short_id(self) -> str:
```

Comment:
Ids already have established format which I didnt want to change as it seems to be used on multiple place. Therefore I have introduced a new property. Or maybe I misunderstood.

Reply:
I was under impression that for all uses of short_id there is `{self.event.id} @`  prefix used anyway but you're right, it is not like that. Because (1) there is another prefix based on Job type (E, J, S...) and (2) different form of id is used in file names. So yeah, please disregard my comment about it - it is clearly needed.